clamp/clamp.tex

```diff
@@ -0,0 +1,341 @@
+    
+    %\[
+     %   |h_{\clamp_{L, U}(v)}(U) - h_{\clamp_{L, U}(w)}(U)| = |h_v(U) - h_w(U)| + |h_{\clamp_{L, U}(v)}(z) - h_{\clamp_{L, U}(w)}(z)|.
+    %\]
+    
+    The same argument applies whenever $z < L$. 
+    
+    \silvia{The first subcase discussed here, i.e., when $k_v = k_w$, is also proven by the triangle inequality expression above, but it seemed clean to separate the case where the total sum remains invariant.}
+    %Therefore, we conclude that, in this case,
+    %\[
+     %   ||h_{\clamp_{L, U}(v)} - h_{\clamp_{L, U}(w)}||_1 - |h_v(z) - h_w(z)|
+     %\leq |h_{\clamp_{L, U}(v)}(U) - h_{\clamp_{L, U}(v)}(U)|. 
+    %\]
+    
+    \item $z \in (L, U)$: then, $\clamp(z) = z$. Since $h_v(z) = h_{\function(v)}(z)$ and $h_v(w) = h_{\function(w)}(z)$, it follows that $|h_v(z) - h_w(z)| = |h_{\function(v)}(z) - h_{\function(w)}(z)|$. Hence the histogram count, i.e., the quantity
+     \[
+        \sum_z |h_{\function(v)}(z) - h_{\function(w)}(z)|
+    \]
+    remains invariant.
+    
+    % \item $z \leq L$: then, $\clamp_{L, U}(z) = L$. Then, $\clamp_{L, U (v)}(z) = \clamp_{L, U (w)}(z) = U$, and hence $|h_{\clamp_{L, U}(v)}(z) - h_{\clamp_{L, U}(w)}(z)| = 0$.
+    
+    \item $z = U$ or $z = L$: then, in the former case, $\clamp(z) = U$. If $z \in \MultiSet(v) \cup \MultiSet(w)$ with the same multiplicity in both multisets, then the histogram count remains invariant under the addition of element $z$. Otherwise, if $z \in \MultiSet(v) \setminus \MultiSet(w)$, or if $z$ is in their union but with different multiplicity, then element $z$ can increase the quantity $|h_{\function(v)}(U) - h_{\function(w)}(U)|$ by at most $|h_v(z)-h_w(z)|$, following the same reasoning with the triangle inequality as in case~2.
+    
+    The same argument applies whenever $z = L$.
+    % Probably simplify to add this case in the others.
+\end{enumerate}
+
+%\silvia{Make cases more concise; avoid repetitions (esp. to try to make proofs ``automatic")}
+
+By aggregating the three cases above, we conclude that
+\[
+\sum_z |h_{\function(v)}(z) - h_{\function(w)}(z)| \leq \sum_z |h_v(z) - h_w(z)|.
+\]
+By the initial assumptions, we recall that $\din \leq \dout$, and that $v, w$ are $\din$-close. Then,
+\[
+\sum_z |h_{\function(v)}(z) - h_{\function(w)}(z)| \leq \sum_z |h_v(z) - h_w(z)| \leq \din \leq \dout.
+\]
+Therefore, 
+\[
+|\MultiSet(\function(v)) \Delta \MultiSet(\function(w))| \leq \dout,
+\]
+as we wanted to show.
+\end{proof}
+
+\silvia{Maybe add domain of $z$ below the sum?}
+%\grace{Great, very rigorous}
+% correct domains above?
+
+% More proofs? Mike: Most proofs give conditions for which they are valid on. This mostly captures domains and metrics. You need a proof for the relation-- the function pseudocode, domains and metrics are all necessary to qualify the proof. I guess it may also make sense to have a proof for the output domain.
+
+% Not sure: separate into vec and sensitivity?
+
```

Comment:
At this point in the PDF (page 6) we see "OLD VERSION BEFORE PROF. VADHAN’S CORRECTIONS – IGNORE". I assume before this is merged we want all of this gone from the PDF.

Reply:
Yes, I have now uploaded a PDF that contains only the relevant pages